module/VuFind/src/VuFind/Log/LoggerFactory.php

```diff
@@ -58,173 +79,132 @@ class LoggerFactory implements FactoryInterface
         if ($pos > 0) {
-            $file = substr($config, 0, $pos);
-            $error_types = substr($config, $pos + 1);
+            $file = substr($configString, 0, $pos);
+            $error_types = substr($configString, $pos + 1);
         } else {
-            $file = $config;
+            $file = $configString;
             $error_types = '';
         }
 
-        // Make Writers
-        $filters = explode(',', $error_types);
-        $writer = new Writer\Stream($file);
-        $this->addWriters($logger, $writer, $filters);
+        $baseFileHandler = new StreamHandler($file, LogLevel::DEBUG, false);
+
+        // Use the generic addHandlers method to configure and add the filtered handlers
+        $this->addHandlers($monologLogger, $baseFileHandler, $error_types);
     }
 
     /**
-     * Configure Slack writers.
+     * Configure Mail handler.
      *
-     * @param Logger             $logger    Logger object
-     * @param ContainerInterface $container Service manager
-     * @param Config             $config    Configuration
+     * @param MonologLogger      $monologLogger The Monolog logger instance to add handlers to.
+     * @param Config             $config        Configuration
+     * @param ContainerInterface $container     Service manager
      *
      * @return void
      */
-    protected function addSlackWriters(
-        Logger $logger,
-        ContainerInterface $container,
-        Config $config
-    ) {
-        $options = [];
-        // Get config
-        [$channel, $error_types] = explode(':', $config->Logging->slack);
-        if ($error_types == null) {
-            $error_types = $channel;
-            $channel = null;
-        }
-        if ($channel) {
-            $options['channel'] = $channel;
-        }
-        if (isset($config->Logging->slackname)) {
-            $options['name'] = $config->Logging->slackname;
-        }
-        $filters = explode(',', $error_types);
-        // Make Writers
-        $writer = new Writer\Slack(
-            $config->Logging->slackurl,
-            $container->get(\VuFindHttp\HttpService::class)->createClient(),
-            $options
-        );
-        $writer->setContentType('application/json');
-        $formatter = new \Laminas\Log\Formatter\Simple(
-            '*%priorityName%*: %message%'
+    protected function addMailHandler(MonologLogger $monologLogger, Config $config, ContainerInterface $container): void
+    {
+        $parts = explode(':', $config->Logging->email);
+        $email = $parts[0];
+        $error_types = $parts[1] ?? '';
+
+        $mailHandler = new MailHandler(
+            $email,
+            'VuFind Log Message',
+            $this->getEmailSenderAddress($config),
+            $container->get(Mailer::class)
         );
-        $writer->setFormatter($formatter);
-        $this->addWriters($logger, $writer, $filters);
+        $deduplicationHandler = new DeduplicationHandler($mailHandler);
```

Comment:
I think it's worth adding a comment here to explain the purpose/intent of this:


```suggestion
        // Do not send an email for each individual log message; instead, bundle them with DeduplicationHandler:
        $deduplicationHandler = new DeduplicationHandler($mailHandler);
```